atlasdb-api/src/main/java/com/palantir/atlasdb/timestamp/DbTimeLockFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timestamp;
+
+import java.util.Optional;
+
+import com.palantir.atlasdb.config.DbTimestampCreationSetting;
+import com.palantir.atlasdb.config.LeaderConfig;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TimestampSeriesProvider;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.timestamp.ManagedTimestampService;
+
+/**
+ * See {@link com.palantir.atlasdb.spi.AtlasDbFactory}. A {@link DbTimeLockFactory} is an extension of an
+ * AtlasDbFactory that is expected to make suitable decisions around certain parameters to create a raw key-value
+ * service, and it also supports extracting the timestamp series known about by the underlying database.
+ */
+public interface DbTimeLockFactory {
```

Comment:
Did we not extend `AtlasDbFactory` because then we run into having two factories of same type? Not a complaint, just asking for a friend :P

Reply:
I didn't even think of that, actually - I went with the standard "prefer composition over inheritance" design principle here. You're right that that would have been a problem with the inheritance model...

Some of it, though, is also to restrict the visibility to users. I don't want people to try getting the timestamp invalidator, or calling createRawKvs with some of the other config options set.